Lib/logging/__init__.py

```diff
@@ -361,6 +366,18 @@ def __init__(self, name, level, pathname, lineno,
         else:
             self.process = None
 
+    @property
```

Comment:
May look out of place compared to the other instance variables within `LogRecord`. 
I was thinking this approach could be a good idea, so that it is evaluated lazily.
Perhaps this could be extended to some of the other fields in a separate PR?

Though, I wasn't sure if it's important that they are evaluated specifically at the point of a LogRecord being initialised.

Reply:
Have realised that the `@property` addition has actually broken the formatting. Also highlights a need for me to add a test surrounding this.